test/qa_buffer.cpp

```diff
@@ -199,7 +199,7 @@ const boost::ut::suite UserApiExamples = [] {
             auto lambda = [](auto w) { // test writer generating consecutive samples
                 static int offset = 1;
                 std::iota(w.begin(), w.end(), offset);
-                offset += w.size();
+                offset += static_cast<int>(w.size());
```

Comment:
N.B. `auto w` is actually `int32_t w` -> cast should probably be `static_cast<int32_t>(w.size());`

Reply:
I think here everything is correct. 'w' is 'std::span<int>', but I cast 'w.size()' to be the same type as offset, which is 'int'. One of the possibility is to declare 'offset' as 'std::size_t'.

Reply:
Maybe 'std::size_t' would be easier and lowering the cognitive complexity ...